### PR TITLE
feat(build): Upgrade to Webpack v4

### PR DESCRIPTION
**Pull request by mbfisher**
BREAKING CHANGE:

- Requires NodeJS v6.11.5.
- Webpack config schema has breaking changes which `dangerouslySetWebpackConfig` users should review.
- Replaces `extract-text-webpack-plugin` with `mini-css-extract-plugin`.

re #125

### CODE REVIEW

**Comment by jahredhope** [comment]
Might be worth reviewing some of the dependencies after this PR, they shouldn't be a problem but worth knowing we are a bit behind on a few things.
screen shot 2018-08-09 at 11 52 13 am

**Comment by mbfisher** [comment]
@jahredhope I was purposefully conservative on updates - don't want to break any more than we have to!

**Comment by mbfisher** [comment]
On the `start-ssr` item; I'm getting 2 issues running it in my project:

image

1. `http://talent.seek.com.au.local:undefined`
2. Can't resolve `__sku_alias__serverEntry`

**Comment by mbfisher** [comment]
However, I get the same with `sku@latest`, so maybe it's something I'm doing in my project. The E2E test is passing...

image

**Comment by markdalgleish** [comment]
@mbfisher have you defined `entry.server` in your sku config?

**Comment by mbfisher** [comment]
@markdalgleish No 😅. Now I've dug into it I hadn't done any of the things I needed to! With a basic `serverEntry` the alias error goes away 👍.

**Comment by mbfisher** [comment]
Build's failing on CI though 🤦‍♂️

**Comment by mbfisher** [comment]
I think we're getting intermittent CI failures because the tests are starting to take around 35s to run, so I've bumped up `jest.setTimeout`.

File: config/webpack/webpack.config.js [hunk 1]
```diff
@@ -136,7 +137,10 @@ const buildWebpackConfigs = builds.map(
       .value();
 
     const internalJs = [paths.src, ...paths.compilePackages];
-    const entry = [paths.clientEntry];
+    const resolvedPolyfills = polyfills.map(polyfill => {
+      return require.resolve(polyfill, { paths: [process.cwd()] });
+    });
+    const entry = [...resolvedPolyfills, paths.clientEntry];
```
Comment (markdalgleish) [inline config/webpack/webpack.config.js]: Is this change related to the webpack 4 upgrade? Seems unrelated—do we know why this was changed?

Comment (syakiruzair) [reply config/webpack/webpack.config.js]: I cannot remember this change, @mbfisher . But if I did, it should be to clean it up as what @markdalgleish mentioned.

Comment (mbfisher) [reply config/webpack/webpack.config.js]: 👍

Comment (hscgavin) [reply config/webpack/webpack.config.js]: 🤔 how come `entry = [...devServerEntries, ...resolvedPolyfills, paths.clientEntry]`?
should be `entry = [ ...resolvedPolyfills, ...devServerEntries, paths.clientEntry]` 
☝️  @markdalgleish

Comment (mbfisher) [reply config/webpack/webpack.config.js]: @hscgavin good point

Comment (hscgavin) [reply config/webpack/webpack.config.js]: the order here is important 👇 

>  it's important for polyfills that modify the global environment to be loaded before all other code. To address this, the polyfills option allows you to provide an array of modules to import before any other code is executed.

**Comment by mbfisher** [comment]
Get very inconsistent test results with timeouts now - the current commit has passed the [PR build](https://travis-ci.org/seek-oss/sku/builds/414353415) but failed the [push build](https://travis-ci.org/seek-oss/sku/builds/414353444).
